crates/core/src/server.rs

```diff
@@ -182,32 +182,28 @@ pub mod local_node {
 pub mod network_node {
     use std::net::SocketAddr;
 
-    use libp2p_identity::Keypair;
     use tower_http::trace::TraceLayer;
 
     use crate::{
-        client_events::websocket::WebSocketProxy, dev_tool::NodeConfig, local_node::PeerCliConfig,
-        DynError,
+        client_events::websocket::WebSocketProxy, config::Config, dev_tool::NodeConfig,
+        transport::TransportKeypair, DynError,
     };
 
     use super::{http_gateway::HttpGateway, serve};
 
-    pub async fn run_network_node(
-        config: PeerCliConfig,
-        socket: SocketAddr,
-    ) -> Result<(), DynError> {
+    pub async fn run_network_node(config: Config, socket: SocketAddr) -> Result<(), DynError> {
         let (gw, gw_router) = HttpGateway::as_router(&socket);
         let (ws_proxy, ws_router) = WebSocketProxy::as_router(gw_router);
         serve(socket, ws_router.layer(TraceLayer::new_for_http()));
 
-        let mut node_config = NodeConfig::new();
+        let mut node_config = NodeConfig::new(config);
         node_config.with_ip(socket.ip()).with_port(socket.port());
 
-        let private_key = Keypair::generate_ed25519();
+        let private_key = TransportKeypair::new();
```

Comment:
so we want to be able to load this is the config optionally or create a new one otherwise